### ISSUE TITLE
Decimal String not adhering to standard?

### ISSUE THREAD

**Issue by edespong** [fo-dicom/fo-dicom#1793]
**Describe the bug**
The VR Decimal String, DS, is defined as:

```
A string of characters representing either a fixed point number or a floating point 
number. A fixed point number shall contain only the characters 0-9 with an optional 
leading "+" or "-" and an optional "." to mark the decimal point. A floating point 
number shall be conveyed as defined in ANSI X3.9, with an "E" or "e" to indicate the 
start of the exponent. Decimal Strings may be padded with leading or trailing spaces. 
Embedded spaces are not allowed.
```

There is no mention that there cannot be leading zeros. Since there is no definition of what a fixed point number is, should not a string such as "030" be valid? 

**Expected behavior?**
The string "030" would be considered valid by the `ValidateDS` in `DicomValidation`?

I am encoutering real-world examples that have these kinds of values.

**Comment by mrbean-bremen** [comment]
I think you are right - I could not find any mention that this is not allowed, and any regex that checks for floating point numbers I could find allows for leading zeros.
Quite easy to fix...

**Comment by gofal** [comment]
Well, I also agree, that this seems to be a valid DS.

**Comment by gofal** [comment]
I will move comment from the PR to this issue: 
The definition of DS says "A fixed point number shall contain only the characters 0-9 with an optional leading "+" or "-" and an optional "." to mark the decimal point". it does not say that it shall contain at least one character 0-9. So would the text "." be a valid DS?

**Comment by mrbean-bremen** [comment]
That is a good question. I would say no by gut feeling, but still couldn't find an official definition. Unofficially, there are several regexes that do not allow this, for example [this one](https://regex101.com/r/ooAv7x/1) which has a good explanation in the comments.
Our own regex is more permissive and allows that, but I would tend to change that...

**Comment by mrbean-bremen** [comment]
After more reading and thinking, I actually changed my mind. 
I'm now inclined to remain more permissive. There seems to be no real standard in the textual representation of floating point numbers, at least concerning the presence of digits around the dot. The representation I linked does not allow for something like `123.`, which I would consider valid. Given that all these strings (including a single dot) unambiguously define numbers, and the DICOM standard says nothing about it, I think we can leave the regex as permissive as it is now.

**Comment by mrbean-bremen** [comment]
@gofal: I suggest to either leave the regex as is in the PR, or change the first part from `"^[+-]?((\d*)([.]\d*)?|[.]\d+)` to something like `"^[+-]?((\d+(\.\d*)?)|(\.\d+))`, which would not allow a single dot, but would allow a leading or trailing dot. What do you think?

**Comment by gofal** [comment]
I would prefer that variant, that is dicom-conform but also makes sure that the generated data can be processed correctly.
If you generate a file with only the dot as a DS value and then open this file in other tools/applications, are they then able to parse this value, or would it cause issues. I will give it a try.

**Comment by gofal** [comment]
I tried Orthanc, ClearCanvas and MicroDicom, they all three do not parse "." as a valid DS number.
e.g. if I set a value to "0.0" or "0." or ".0", then the value is displayed as "0". but if I set the value to ".", then microdicom only shows "."
Orthanc refuses to process a file where I set the WindowCenter to "." and want to upload it via OrthancExplorer.
And ClearCanvas ignores the WindowCenter "." and estimates the Windowing from histrogram.

I stop evaluating now, and I would recommand, to throw a validation exception, if a user wants to add "." as a DS value.
